platform/commonUI/general/src/ui/TreeNodeView.js

```diff
@@ -81,13 +81,15 @@ define([
         }
 
         this.activeObject = domainObject;
-
-        if (domainObject && domainObject.hasCapability('composition')) {
-            $(this.toggleView.elements()).removeClass('no-children');
-        } else {
-            $(this.toggleView.elements()).addClass('no-children');
+        if (domainObject && domainObject.hasCapability('adapter')) {
+            var obj = domainObject.useCapability('adapter');
```

Comment:
@aayusharora - an easier way to do this is to check if domainObject.hasCapability('composition'), or check the model for a composition value. That way you don't need to pass around the openmct object.

Reply:
That will only work where the composition is defined on the object model, it won't work for composition providers. Using `openmct.composition.get()` is more reliable here.

Reply:
Hi @dtailor90 I discussed using hasCapability earlier and discussed it with @akhenry on the issue  #1826
As he stated,
Some objects store their composition in a composition array on the object itself, but this is not universal. There are also 'composition providers' which can provide asynchronous composition. Eg. if an external service needs to be queried to fetch the children of an object.

The most reliable way to test whether an object supports composition is to use the composition API. The Composition API can be queried for composition and will return undefined if it does not support composition at all. eg.

// Because we are dealing with legacy code, but will be using the new API, we need to convert to a new style object.
var newStyleObject = domainObject.useCapability('adapter');
var hasComposition = openmct.composition.get(newStyleObject) !== undefined;
As well as checking the .composition attribute, this method will also look for composition providers associated with this object.

The openmct API is exposed to controllers etc via an Angular service called 'openmct' service, so it can be injected directly by defining a dependency in the associated bundle.js

As a result, I thought of using openmct as a service. Please check if I am doing it correctly.

Thanks :)